ng-dev/utils/git/github.ts

```diff
@@ -65,6 +66,59 @@ export class AuthenticatedGithubClient extends GithubClient {
   async graphql<T extends GraphqlQueryObject>(queryObject: T, params: RequestParameters = {}) {
     return (await this._graphql(query(queryObject).toString(), params)) as T;
   }
+
+  /**
+   * Retrieve a combined listing of the results for a refs statuses and checks.
+   */
+  async getCombinedChecksAndStatusesForRef(
+    params: RestEndpointMethodTypes['checks']['listForRef']['parameters'] &
+      RestEndpointMethodTypes['repos']['getCombinedStatusForRef']['parameters'],
+  ) {
+    const {data: checkResults} = await this.checks.listForRef(params);
+    const {data: statusResults} = await this.repos.getCombinedStatusForRef(params);
+
+    const results = [
+      ...checkResults.check_runs.map((result) => ({
+        type: 'check',
+        name: result.name,
+        result: result.status === 'completed' ? result.conclusion! : result.status,
+        url: result.details_url,
+        check: result,
+      })),
+      ...statusResults.statuses.map((result) => ({
+        type: 'status',
+        name: result.context,
+        result: result.state,
+        description: result.description,
+        url: result.target_url,
+        status: result,
+      })),
+    ];
+
+    return {
+      result: results.reduce(
```

Comment:
Nit: Probably more readable to compute in a for loop and continue early if result === failing.

Reply:
Left a TODO